internal/archive/archive.go

```diff
@@ -257,6 +257,10 @@ func WriteZipToTar(tw *tar.Writer, srcZip, basePath string, uid, gid int, mode i
 			}
 		}
 
+		if overwritePermissions(f.FileHeader) {
```

Comment:
Can this check be done at the top of the loop? Am I understanding it right that *any* `FAT` fileheader would cause the global mode to be overwritten?

Reply:
That is one way of doing it, but we could also do it on a file by file basis, though I'm not sure how much sense that would make.

Reply:
K yeah I'm not that aware of the internals of handling `FAT`, but I want to be sure that I understand the intention. Let me phrase what this code is doing and you can correct me if I'm wrong:

if we identify a file as `FAT`, we overwrite the `mode` for the entirety of the rest of the function call.

That sound right?

Reply:
That is what I'm currently doing. I was opening it up whether I should be doing that, though I would assume that if one file in a zipped dir is a certain format, I should be able to extrapolate that all files are in that format.

Reply:
Gotcha, yeah that's where my knowledge stops.

Reply:
FWIW, I think we're doing the right thing here. A bitwise operation on data we already have present seems like a very small price to pay for the sake of correctness. Each file, could technically, yet not practically, have a different "creator".

Reply:
@jromero Are you arguing in favor of checking/setting the `mode` for every file, or for doing it once for the zipped dir?

Reply:
> checking/setting the mode for every file

Reply:
I pushed code that does that - but there's no real way of testing it. Do you still think it makes sense to check file by file?

Reply:
Thanks. I think this is as good as one could wish for. I appreciate the quick change.